packages/next/src/server/pipe-readable.ts

```diff
@@ -69,9 +71,15 @@ export async function pipeReadable(
   try {
     while (true) {
       const { done, value } = await reader.read()
-      readerDone = done
 
-      if (done || writableClosed) {
+      if (done) {
+        state |= kReaderDone
+        break
+      } else {
+        state &= kReaderDone
```

Comment:
This should be `state ^= kReaderDone` or `state &= ~kReaderDone` as we need to set the `readerDone` state to be false here, right?

Reply:
Yes, you're right, forgot the ~.